waiter/src/waiter/config.clj

```diff
@@ -0,0 +1,37 @@
+;;
+;; Copyright (c) Two Sigma Open Source, LLC
+;;
+;; Licensed under the Apache License, Version 2.0 (the "License");
+;; you may not use this file except in compliance with the License.
+;; You may obtain a copy of the License at
+;;
+;;  http://www.apache.org/licenses/LICENSE-2.0
+;;
+;; Unless required by applicable law or agreed to in writing, software
+;; distributed under the License is distributed on an "AS IS" BASIS,
+;; WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+;; See the License for the specific language governing permissions and
+;; limitations under the License.
+;;
+(ns waiter.config)
+
+(def ^:private config-promise (promise))
+
+(defn initialize-config
+  "Initializes the config."
+  [config-value]
+  (if-not (realized? config-promise)
+    (deliver config-promise config-value)
+    (throw (IllegalStateException. "The config has already been initialized"))))
+
+(defn- check-config-initialized
```

Comment:
I like the word `require` better than `check` here. We could also use Clojure's precondition support.

Reply:
Changed to use pre-conditions.